src/data/library.ts

```diff
@@ -151,6 +122,85 @@ class Library {
     });
   };
 
+  // NOTE this is a high-level public API (eposed on ExcalidrawAPI) with
+  // a slight overhead (always restoring library items). For internal use
+  // where merging isn't needed, use `library.setLibrary()` directly.
+  updateLibrary = async ({
+    libraryItems,
+    prompt = false,
+    merge = false,
+    openLibraryMenu = false,
```

Comment:
This I feel can be exposed as a different API in excal api instead and its not related to update library

Reply:
so far opening the library menu is/should be tied to updating/importing only, so splitting the API would only complicate things

Reply:
`updatingLibrary` refers to updating lib items in some way (adding/removing/replacing) but opening the lib menu is an action, more specifically a UI action so if we want it to be controlled by the host then I think separate API would be better. And if we want to always open the library then we can keep it here and remove the param `openLibararyMenu` altogether

Reply:
The `updateLibrary` is a high-level API, same as `updateScene`, which also takes `appState` (which can alter UI state) and `commitToHistory`. It's about usefulness and semantics. Since opening the library menu is currently tied to updating the library, it makes sense to expose it.

The alternative is to force host apps to do an extra:

```
excalidrawAPI.updateScene({ appState: { isLibraryOpen: true } });
```

I don't care much, but I think the argument here and in your other comments stems from wanting to make this API low-level, which we don't need to.